src/modules/load_mon/load_mon.cpp

```diff
@@ -247,12 +247,13 @@ void LoadMon::_stack_usage()
 
 		task_stack_info_s task_stack_info = {};
 
-		if (system_load.tasks[task_index].valid && system_load.tasks[task_index].tcb->pid > 0) {
+		if (system_load.tasks[task_index].valid && (system_load.tasks[task_index].tcb->pid > 0)) {
 
 			stack_free = up_check_tcbstack_remain(system_load.tasks[task_index].tcb);
 
-			strncpy((char *)task_stack_info.task_name, system_load.tasks[task_index].tcb->name,
-				task_stack_info_s::MAX_REPORT_TASK_NAME_LEN);
+			static_assert(sizeof(task_stack_info.task_name) == CONFIG_TASK_NAME_SIZE,
+				      "task_stack_info.task_name must match NuttX CONFIG_TASK_NAME_SIZE");
+			strncpy((char *)task_stack_info.task_name, system_load.tasks[task_index].tcb->name, CONFIG_TASK_NAME_SIZE);
```

Comment:
I think newer compilers are unhappy if you copy `sizeof` the char array and you need to copy `sizeof - 1` and make sure the end byte is `\0`.

Reply:
Updated.